hail/python/hailtop/batch_client/validate.py

```diff
@@ -0,0 +1,316 @@
+import re
+
+# rough schema (without requiredness, value validation):
+# jobs_schema = [{
+#   'always_run': bool,
+#   'attributes': {str: str},
+#   'callback': str,
+#   'command': [str],
+#   'env': [{
+#     'name': str,
+#     'value': str
+#   }],
+#   'image': str,
+#   'input_files': [str],
+#   'job_id': int,
+#   'mount_docker_socket': bool,
+#   'output_files': [str],
+#   'parent_ids': [int],
+#   'pvc_size': str,
+#   'resoures': {
+#     'memory': str,
+#     'cpu': str
+#   },
+#   secrets: [{
+#     'namespace': str,
+#     'name': str,
+#     'mount_path': str
+#   }],
+#   service_account_name: str
+# }]
+
+JOB_KEYS = {
+    'always_run', 'attributes', 'callback', 'command', 'env', 'image', 'job_id', 'mount_docker_socket', 'parent_ids', 'resources', 'secrets', 'service_account_name'
+}
+
+ENV_VAR_KEYS = {'name', 'value'}
+
+SECRET_KEYS = {'namespace', 'name', 'mount_path'}
+
+RESOURCES_KEYS = {'memory', 'cpu'}
+
+K8S_NAME_REGEXPAT = r'[a-z0-9](?:[-a-z0-9]*[a-z0-9])?(?:\.[a-z0-9](?:[-a-z0-9]*[a-z0-9])?)*'
+K8S_NAME_REGEX = re.compile(K8S_NAME_REGEXPAT)
+
+MEMORY_REGEXPAT = r'[+]?(?:[0-9]*[.])?[0-9]+[KMGTP]?'
+MEMORY_REGEX = re.compile(MEMORY_REGEXPAT)
+
+CPU_REGEXPAT = r'[+]?(?:[0-9]*[.])?[0-9]+[u]?'
```

Comment:
Are we changing which CPU we allow now? I thought the unit could be an "m".

Reply:
Oops, typo, thanks.